src/daemon/daemon.cpp

```diff
@@ -1742,25 +1757,26 @@ try // clang-format on
             timestamp->set_nanos(snapshot->get_creation_timestamp().time().msec() * 1'000'000);
         };
 
-        if (const auto& it = instance_snapshots_map.find(name);
-            it == instance_snapshots_map.end() || it->second.empty())
-        {
-            for (const auto& snapshot : vm.view_snapshots())
-                get_snapshot_info(snapshot);
-        }
-        else
+        const auto& it = instance_snapshots_map.find(name);
+        const auto& [pick, all] = it == instance_snapshots_map.end() ? SnapshotPick{{}, true} : it->second;
+
+        try
         {
-            for (const auto& snapshot : it->second)
+            if (all)
             {
-                try
-                {
-                    get_snapshot_info(vm.get_snapshot(snapshot));
-                }
-                catch (const NoSuchSnapshot& e)
-                {
-                    add_fmt_to(errors, e.what());
-                }
+                for (const auto& snapshot : pick)
+                    vm.get_snapshot(snapshot); // verify validity of any snapshot name requested separately
```

Comment:
Why not use `verify_snapshot_picks()` in `info` as well?

Reply:
Because we're fetching for an individual VM here, whereas `verify_snapshot_picks` checks all snapshot arguments. We could call it before calling this lambda, but we'd be calling `VM::get_snapshot` twice. 

Unlike `delete`, `info` does not change instances or snapshots, so it's fine if we detect a bad arg and abort the operation in the middle. I realize that fetching a bunch of info and then throwing it away is more wasteful than fetching each snapshot twice. However, the former only occurs when there is an error, whereas the latter would take place every time. I thought it preferable to optimize the successful path.

Reply:
ah, I see. Alright, I think that's ok.